### PR TITLE
DISPATCH-1749: Improve test framwork router.wait_ready function

### PR DESCRIPTION
**Pull request by ChugR**
This patch adds new check for testing that a router is ready.

  * Optional. Enabled by default.
  * Locates "SERVER (notice) Operational" in a log file.
  * Fails test if log settings don't support seeing log message.
  * Fails wait_ready if message never seen.
  * Additional test runs before existing port and connector probes.

### CODE REVIEW

File: tests/system_test.py [hunk 1]
```diff
@@ -421,29 +437,71 @@ def attributes(e, level):
...
             return "".join(["%s {\n%s}\n"%(n, attributes(p, 1)) for n, p in self])
 
-    def __init__(self, name=None, config=Config(), pyinclude=None, wait=True,
+    def __init__(self, name=None, config=Config(), pyinclude=None, wait=True, wait_for_server_log=True,
                  perform_teardown=True, cl_args=None, expect=Process.RUNNING):
         """
         @param name: name used for for output files, default to id from config.
         @param config: router configuration
         @keyword wait: wait for router to be ready (call self.wait_ready())
+        @param wait_for_server_log Use server log check for "wait ready"
```
Comment (jiridanek) [inline tests/system_test.py:446]:
```suggestion
        @param wait_for_server_log: use server log check for "wait ready"
```

File: tests/system_test.py [hunk 2]
```diff
@@ -421,29 +437,71 @@ def attributes(e, level):
...
+        @param perform_teardown
+        @param cli_args
+        @param expect
```
Comment (jiridanek) [inline tests/system_test.py:449]: Is this good for anything? mentioning the parameters but not giving any description?

File: tests/system_test.py [hunk 3]
```diff
@@ -421,29 +437,71 @@ def attributes(e, level):
             # top level list of tuples ('section-name', dict)
             return "".join(["%s {\n%s}\n"%(n, attributes(p, 1)) for n, p in self])
 
-    def __init__(self, name=None, config=Config(), pyinclude=None, wait=True,
+    def __init__(self, name=None, config=Config(), pyinclude=None, wait=True, wait_for_server_log=True,
```
Comment (jiridanek) [inline tests/system_test.py:440]: There are now three ways of waiting for server readiness. Wait for open port, wait for `/healtz` endpoint (if libwebsockets is enabled for compilation) or wait for special log line. It feels kind of confusing, what is the difference, when should I do what, and so on.

**Comment by jiridanek** [comment]
Typo in Jira name in commit message.

File: tests/system_test.py [hunk 4]
```diff
@@ -421,29 +437,71 @@ def attributes(e, level):
...
         # setup log and debug dump files
         self.dumpfile = os.path.abspath('%s-qddebug.txt' % name)
         self.config.sections('router')[0]['debugDumpFile'] = self.dumpfile
+
+        # setup default logging
+        # sense where server info logs are found
+        # fail if server polling log ready and log setup is insufficient
+        self.logfile = None
+        self.server_logfile = None
+        self.server_search_pattern = "SERVER (notice) Operational"
```
Comment (jiridanek) [inline tests/system_test.py:466]: I searched and the `QD_LOG_NOTICE, "Operational, %d Threads Running (process ID %ld)",` line seems to be the first thing the router prints after it registers signal handlers in `static void main_process()`. The sigterm handler catches that signal and essentially converts that to a clean exit.

Makes sense. The "crash" is detected by test code if SIGTERM is delivered before dispatch sets a handler for it.

**Comment by ChugR** [comment]
Thanks for the comments @jiridanek but I'm withdrawing this PR. The new code presented here is itself open to problems as yet unseen.

**Pull request closed without merging**